### PR TITLE
Update to latest Ignite

### PR DESCRIPTION
**Pull request by kateinkim**
I worked through [the tutorial](https://shift.infinite.red/creating-a-trivia-app-with-ignite-bowser-part-1-1987cc6e93a1) and applied the changes to a fresh Ignite install fixing the parts that no longer apply with the new Ignite.

### CODE REVIEW

File: app/screens/question/question-screen.tsx [hunk 1]
```diff
@@ -50,20 +51,19 @@ const CHECK_ANSWER: ViewStyle = {
...
-  // Are we refreshing the data
   const [refreshing, setRefreshing] = useState(false)
 
   // Pull in one of our MST stores
   const { questionStore } = useStores()
   const { questions } = questionStore
 
   useEffect(() => {
-    fetchQuestions()
+    setTimeout(fetchQuestions, 1)
```
Comment (robinheinze) [inline app/screens/question/question-screen.tsx:63]: Can you add a comment here with the link to the Ignite issue?

File: app/services/api/api.ts [hunk 2]
```diff
@@ -4,23 +4,20 @@ import { ApiConfig, DEFAULT_API_CONFIG } from "./api-config"
...
 import { QuestionSnapshot } from "../../models/question/question"
 import * as uuid from "react-native-uuid"
-import { decodeHTMLEntities } from "../../utils/html-decode"
 
-const API_PAGE_SIZE = 10
+const API_PAGE_SIZE = 50
 
 const convertQuestion = (raw: any): QuestionSnapshot => {
-  const id = uuid.v1().toString()
-  const decodedQuestion = decodeHTMLEntities(raw.question)
```
Comment (robinheinze) [inline app/services/api/api.ts:13]: Is this HTML decoding no longer needed?

File: app/services/api/api.ts [hunk 3]
```diff
@@ -67,11 +64,13 @@ export class Api {
   }
 
   /**
-   * Gets a list of trivia questions.
+   * Gets a list of users.
```
Comment (robinheinze) [inline app/services/api/api.ts]: Oops, overwrote the comment here :)

File: app/services/api/api.ts [hunk 4]
```diff
@@ -90,3 +89,7 @@ export class Api {
     }
   }
 }
+
+/**
+ * Gets a single user by ID
+ */
```
Comment (robinheinze) [inline app/services/api/api.ts]: This comment can be removed as well

File: app/services/api/api.types.ts [hunk 5]
```diff
@@ -1,4 +1,16 @@
 import { GeneralApiProblem } from "./api-problem"
+import { Character } from "../../models/character/character"
 import { QuestionSnapshot } from "../../models/question/question"
 
+export interface User {
```
Comment (robinheinze) [inline app/services/api/api.types.ts]: Shouldn't need these types either

File: app/models/character-store/character-store.test.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { CharacterStoreModel } from "./character-store"
```
Comment (robinheinze) [inline app/models/character-store/character-store.test.ts]: Do these character / character store models come with Ignite now? We can probably remove them from this app. I can make a note in the blog post text

File: app/screens/question/question-screen.tsx (hunk 1 above)
Comment (kateinkim) [reply app/screens/question/question-screen.tsx:63]: Yep 👍

File: app/services/api/api.ts (hunk 2 above)
Comment (kateinkim) [reply app/services/api/api.ts:13]: I will double check on this

File: app/services/api/api.ts (hunk 3 above)
Comment (kateinkim) [reply app/services/api/api.ts]: Ooops sorry

File: app/services/api/api.ts (hunk 4 above)
Comment (kateinkim) [reply app/services/api/api.ts]: I will clean up those

File: app/models/character-store/character-store.test.ts (hunk 6 above)
Comment (kateinkim) [reply app/models/character-store/character-store.test.ts]: After I opened this PR I realize that. Yes, those come with Ignite. I will delete that too. 🙂

File: app/services/api/api.ts (hunk 2 above)
Comment (kateinkim) [reply app/services/api/api.ts:13]: CleanShot 2021-08-11 at 20 58 55@2x

This is from your blog and it seems decodedQuestion wasn't there already. Maybe it was updated before? However I found this in question-screen.tsx file. See the comment below:

```typescript
const renderQuestion = ({ item }) => {
  const question: Question = item
  return (
    <View style={QUESTION_WRAPPER}>
      {/* decoded here */}
      <Text style={QUESTION} text={decodeHTMLEntities(question.question)} /> 
      <RadioButtons
        options={question.allAnswers}
        onSelection={(guess: string) => onPressAnswer(question, guess)}
        selectedOption={question.guess}
        renderOption={renderAnswer}
      />
      <Button style={CHECK_ANSWER} onPress={() => checkAnswer(question)} text={"Check Answer!"} />
    </View>
  )
}
```

**Comment by kateinkim** [comment]
The ones that were requested to be fixed are all done. However, I noticed this spinner is hanging and not disappearing at all. I don't see why it's behaving this way. It's the same color as the background so it's hard to see but it's definitely there. 



`setRefreshing` is set to false.

```typescript
const fetchQuestions = async () => {
    setRefreshing(true)
    await questionStore.getQuestions()
    setRefreshing(false)
  }
```

**Review by robinheinze — APPROVED** [review]

**Pull request merged**